### PR TITLE
[image_picker] Fix file not found exception

### PR DESCRIPTION
**Pull request by yapcwed**
## Description

This PR adds a check to ensure the directory where scaled images are saved to exist. This issue caused the app to crash on a Xiaomi Mi 9T (FileNotFoundException).

## Related Issues

* [flutter/flutter#46079](https://github.com/flutter/flutter/issues/46079)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by cyanglaz** [comment]
@yapcwed Do you think there is a way to add a test?

**Comment by yapcwed** [comment]
I think it would theoretically be possible with an instrumented test but I wouldn't know how to implement it. It's unfortunate that the [android docs](https://developer.android.com/reference/android/content/Context#getExternalFilesDir(java.lang.String)) don't mention the possibility of an externalFilesDir not existing. In any case I think it is good practice to ensure the parent directories exist before creating the file.

**Comment by yapcwed** [comment]
@cyanglaz Can we please get this merged before there are more conflicts?

**Comment by yapcwed** [comment]
@cyanglaz @mklim Can we please finally get this merged? It's really frustrating having to maintain a fork for such a small and straightforward fix.

**Comment by cyanglaz** [comment]
@yapcwed Maybe add a jUnit test: mock that the parent file doesn't exist when resizing the image, then after resizing, check if it crashes.

**Comment by yapcwed** [comment]
@cyanglaz Apologies for the radio silence. I added the unit test as you suggested. And the plugin indeed crashes without the fix provided in this PR. Could you please do a final review and merge the PR if it's fine? Thank you in advance.

**Comment by cyanglaz** [comment]
@yapcwed Thanks. The new version is not reflected on the pubspec.yaml. could you please add it as well?

**Comment by yapcwed** [comment]
You're right, sorry forgot about that. Pubspec.yaml version updated.

File: packages/image_picker/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.6.3+3
+
+* Android: Fix a crash when `externalFilesDirectory` does not exist
```
Comment (cyanglaz) [inline packages/image_picker/CHANGELOG.md]:
```suggestion
* Android: Fix a crash when `externalFilesDirectory` does not exist.
```

**Comment by yapcwed** [comment]
@cyanglaz Is this ok to be merged now? Would be good otherwise merge conflicts will keep popping up.

**Comment by yapcwed** [comment]
@cyanglaz @mklim Is there anything else I can do to get this fix merged and released?

**Comment by yapcwed** [comment]
@mklim You're welcome. Thank __you__ for reviewing and merging the patch!

**Pull request merged**